File: php/src/Model/WorkerDeviceInput.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

use Fivexer\SDK\Internal\Json;

/**
 * An Expo push token from the native app. Browsers use {@see PushSubscriptionInput} instead.
 *
 * Fluent builder: set what you need, then hand it to the client. {@see self::toArray()} omits
 * every field that was never set, because the API treats an absent field as "leave unchanged"
 * and an explicit null as "clear it".
 */
final class WorkerDeviceInput
{
    public function __construct(
        private readonly string $token,
        /** 'ios' | 'android' | 'unknown' */
        private ?string $platform = null,
    ) {
    }

    public function platform(string $platform): self
    {
        $this->platform = $platform;
        return $this;
    }

    /**
     * Serialise for the wire, omitting unset fields.
     *
     * @return array<string, mixed>
     */
    public function toArray(): array
    {
        return [
            'token' => $this->token,
        ] + Json::compact([
            'platform' => $this->platform ?? 'unknown',
        ]);
    }
}

File: java/src/main/java/io/fivexer/sdk/model/WorkerDeviceInput.java
package io.fivexer.sdk.model;

import io.fivexer.sdk.internal.Json;

/**
 * An Expo push token from the native app. Browsers use {@link PushSubscriptionInput} instead.
 *
 * <p>Fluent builder: set what you need, then hand it to the client. {@link #toJson()} omits
 * every field that was never set, because the API treats an absent field as "leave unchanged"
 * and an explicit null as "clear it".
 */
public class WorkerDeviceInput {
    private String token;
    private String platform;

    public WorkerDeviceInput(String token) {
        if (token == null) {
            throw new IllegalArgumentException("token is required");
        }
        this.token = token;
    }

    public String getToken() { return token; }
    public String getPlatform() { return platform; }

    public WorkerDeviceInput platform(String platform) { this.platform = platform; return this; }

    /** Serialise for the wire, omitting unset fields. */
    public String toJson() {
        return Json.write(this);
    }
}
